apiserver/application/application.go

```diff
@@ -945,6 +945,44 @@ func (api *API) saveRemoteApplication(
 			Scope:     ep.Scope,
 		}
 	}
+
+	// If the a remote application with the same name and endpoints from the same
+	// source model already exists, we will use that one.
+	existingRemoteApp, err := api.backend.RemoteApplication(applicationName)
+	if err != nil && !errors.IsNotFound(err) {
+		return nil, errors.Trace(err)
+	}
+	// We may be able to re-use the existing remote app if it is from the same
+	// model and the endpoints are compatible.
+	if err == nil && existingRemoteApp.SourceModel().Id() == sourceModelTag.Id() {
```

Comment:
maybe we should just fail here if the source tags are different? with a less generic message?